smtp/lib.c

```diff
@@ -78,7 +78,7 @@ luaT_smtpc_request(lua_State *L)
 	if (req == NULL)
 		return luaT_error(L);
 
-	double timeout = 365 * 24 * 3600;
+	double timeout = 10;
```

Comment:
The default timeout is changed here for all usages, not just for tests.

Reply:
Yes, I know this one isn't very good. But 10 seconds timeout for RCPT command seems reasonable to me, something wrong must be happening if we hit it? not sure about 1 year timeout om RCPT command.

Reply:
Sure, infinite timeouts are generally discouraged. But it is not the reason to change it here: it is irrelevant to the issue. And we definitely should not change it silently: without any mention in the commit message (but better move to its own commit).

AFAIU, we should pass the timeout option to test the fix of the error message and something went wrong with this functionality on Linux. If it is so, it looks as a bug, which blocks testing of this change. I suggest to file an issue about this with a small reproducer and tarantool/libcurl/module versions (plus platform: distribution and distro version).

Reply:
I see; my bad. I missed the point I can control timeout from lua code directly.

> If it is so, it looks as a bug, which blocks testing of this change. I suggest to file an issue about this with a small reproducer and tarantool/libcurl/module versions (plus platform: distribution and distro version).

Sure, will do.

Reply:
The issue is smtp_h spamming 502 Not implemented every time it receives empty reply, we just needed return statement after it. Not related to curl timeout in any way.
Fixed.